### PR TITLE
Add outline to scatterplot

### PR DESCRIPTION
**Pull request by mcnuttandrew**
screen shot 2016-10-24 at 4 50 23 pm

Add outline to scatterplot in response to #55

### CODE REVIEW

File: src/layers/core/scatterplot-layer/scatterplot-layer.js [hunk 1]
```diff
@@ -98,10 +110,17 @@ export default class ScatterplotLayer extends Layer {
         })
       ),
       geometry: new Geometry({
-        drawMode: 'TRIANGLE_FAN',
+        drawMode: drawMode || 'TRIANGLE_FAN',
         positions: new Float32Array(positions)
       }),
-      isInstanced: true
+      isInstanced: true,
+      onBeforeRender() {
```
Comment (ibgreen) [inline src/layers/core/scatterplot-layer/scatterplot-layer.js]: In deck.gl v3 it is possible to override Layer.draw(). That enables cleaner handling of gl parameters compared to setting state and using callbacks on the model
 See [ArcLayer](https://github.com/uber/deck.gl/blob/dev/src/layers/core/arc-layer/arc-layer.js#L70).

File: example/layer-examples.js [hunk 2]
```diff
@@ -68,6 +68,8 @@ const ScatterplotLayerExample = props =>
   new ScatterplotLayer({
     data: props.points,
     opacity: 0.5,
+    outlineMode: true,
+    outlineStrokeWidth: 2,
```
Comment (ibgreen) [inline example/layer-examples.js]: I believe that all the other layers that support line width prop just call it `strokeWidth`, so calling it that would be a little more consistent.

File: src/layers/core/scatterplot-layer/scatterplot-layer.js [hunk 3]
```diff
@@ -65,6 +67,16 @@ export default class ScatterplotLayer extends Layer {
   }
 
   willReceiveProps(oldProps, newProps) {
+    if (oldProps.outlineMode !== newProps.outlineMode) {
+      const {gl} = this.state;
+      const model = this._getModel(gl, newProps.outlineMode ? 'LINE_LOOP' : 'TRIANGLE_FAN');
```
Comment (ibgreen) [inline src/layers/core/scatterplot-layer/scatterplot-layer.js]: I'd prefer to use GL constants here. `GL.LINE_LOOP` and `GL.TRIANGLE_FAN`. `GL` can be imported from luma, see other layers on dev (if examples are needed). 

The strings will be looked up and resolved to those constants, but no need to make the reader guess what is happening.

File: src/layers/core/scatterplot-layer/scatterplot-layer.js [hunk 4]
```diff
@@ -65,6 +67,16 @@ export default class ScatterplotLayer extends Layer {
...
+
+      this.state.model = model;
+    }
+    if (newProps.outlineMode) {
```
Comment (ibgreen) [inline src/layers/core/scatterplot-layer/scatterplot-layer.js]: This could be cut if going with `draw`

File: example/layer-examples.js (hunk 2 above)
Comment (contra) [inline example/layer-examples.js]: I think to make it consistent with choropleth (`drawContour`) the `outlineMode` option should be `drawOutline`

**Pull request merged**